### DISCUSSION TITLE
Handling lower case headers in Rack 2.x

### DISCUSSION THREAD

**Discussion by ioquatix** [rack/rack#1915] [General]
https://github.com/rack/rack/blob/03dd490d4e62d83ca148fdf0883d2ce3b191dfee/lib/rack/etag.rb#L37

Some Rack 2.x middleware does not correctly handle lower case headers.

This might cause more pain going forwards as more systems normalise to lower case.

We should probably correctly handle lower case headers. @jeremyevans WDYT?

(For more background see https://github.com/rails/sprockets/issues/746).

**Comment by jeremyevans** [comment]
I'm OK with additional middleware using `headers = Utils::HeaderHash[headers]` in Rack 2.x.

**Reply by ioquatix** [reply]
Okay, I've added this and @tenderlove released 2.2.4 which included this.
